x-pack/plugins/apm/server/lib/transaction_groups/get_metrics.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { merge } from 'lodash';
+import { arrayUnionToCallable } from '../../../common/utils/array_union_to_callable';
+import { Transaction } from '../../../typings/es_schemas/ui/transaction';
+import {
+  TRANSACTION_SAMPLED,
+  TRANSACTION_DURATION,
+} from '../../../common/elasticsearch_fieldnames';
+import {
+  AggregationInputMap,
+  SortOptions,
+} from '../../../typings/elasticsearch/aggregations';
+import { TransactionGroupRequestBase, TransactionGroupSetup } from './fetcher';
+
+interface MetricParams {
+  request: TransactionGroupRequestBase;
+  setup: TransactionGroupSetup;
+}
+
+type BucketKey = string | Record<string, string>;
+
+function withAggs<
+  TRequestBase extends TransactionGroupRequestBase,
+  TInputMap extends AggregationInputMap
+>(request: TRequestBase, aggs: TInputMap) {
+  return merge({}, request, {
+    body: {
+      aggs: {
+        transaction_groups: {
+          aggs,
+        },
+      },
+    },
+  });
+}
+
+export async function getSamples({ request, setup }: MetricParams) {
+  const params = withAggs(request, {
+    sample: {
+      top_hits: {
+        size: 1,
+      },
+    },
+  });
+
+  const sort: SortOptions = [
+    { _score: 'desc' as const }, // sort by _score to ensure that buckets with sampled:true ends up on top
+    { '@timestamp': { order: 'desc' as const } },
+  ];
+
+  const response = await setup.client.search({
+    ...params,
+    body: {
+      ...params.body,
+      query: {
+        ...params.body.query,
+        bool: {
+          ...params.body.query.bool,
+          should: [{ term: { [TRANSACTION_SAMPLED]: true } }],
+        },
+      },
+      sort,
+    },
+  });
+
+  return arrayUnionToCallable(
+    response.aggregations?.transaction_groups.buckets
+  ).map((bucket) => {
```

Comment:
Is this the equivalent of?
```suggestion
  return (response.aggregations?.transaction_groups.buckets ?? []).map((bucket) => {
```

Reply:
no, it's this issue: https://ela.st/union-callable